### DISCUSSION TITLE
pr-pull action fails with "Error: No artifact with the name `bottles` was found!"

### DISCUSSION THREAD

**Discussion by kke** [Homebrew/discussions#4687] [Tap maintenance and Homebrew development] [answered]
The tap in question used to work fine but now I'm getting an error in the Pull Bottles step after labeling the PR with `pr-pull`.

I tried to copy the workflows from a fresh "brew new-tap" invocation, but it didn't help.

An example of a failed run is here: https://github.com/k0sproject/homebrew-tap/actions/runs/5726550099/job/15517219113

The PR is this one: https://github.com/k0sproject/homebrew-tap/pull/20

Any ideas?

**Comment by osalbahr** [comment]
Not sure. Are you able to replicate the same error by manually running `brew` commands?

**Comment by Bo98** [comment, marked as answer]
Probably because your test failed:

```
 ==> FAILED
Full test k0sproject/tap/k0sctl output
  Warning: test failed
  /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/bin/bundle clean
  ==> Testing k0sproject/tap/k0sctl
  ==> /home/linuxbrew/.linuxbrew/Cellar/k0sctl/0.15.3/bin/k0sctl version 2>&1
  Error: k0sproject/tap/k0sctl: failed
  An exception occurred within a child process:
    Minitest::Assertion: Expected /version:\ v0\.15\.3/ to match "version: \ncommit: unknown\n".
```

The reason it didn't fail the whole CI is because `brew test-bot` doesn't really work properly when you manually remove the bottle block like you did (though that's not to say it shouldn't handle this better - I'll have a look). The recommendation is to not touch the bottle block and the `pr-pull` step will replace it.

**Reply by kke** [reply]
Ah ok, so far I've always removed it and it has worked. I now kept the old bottle block and it worked (or rather, successfully failed at the correct step).

Thank you.
